File: 프로그래머스/src/프로그래머스_02262022/콜라츠_추측.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int num)
{
    int answer = 0;
    int cnt = 500;
    long number = (long)num; //이부분..
    while (cnt > 0)
    {
        if (number == 1)
            return answer; // end case
        if (number % 2 == 0)
            number /= 2;
        else
            number = (number * 3) + 1;
        ++answer;
        --cnt;
    }
    return -1;
}

File: 2026/Level1/콜라츠추측.py
# https://school.programmers.co.kr/learn/courses/30/lessons/12943?language=python3
def solution(num):
    answer = 0
    cnt = 0
    while cnt < 500:
        # 2
        if num == 1 :
            return answer
        # 1-1
        if num%2==0:
            num = num /2 
        # 1-2
        else :
            num = num * 3 + 1
        answer +=1
        cnt +=1
    return -1